crates/web-plugins/didcomm-messaging/src/did_rotation/errors.rs

```diff
@@ -0,0 +1,34 @@
+use axum::Json;
+use serde_json::{json, Value};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum RotationError {
+    #[error("Could not deserialize from prior")]
+    DeserializationError,
+    #[error("Could not rotate did unknown issuer")]
+    UnknownIssuer,
+    #[error("Invalid jwt signature on FromPrior value")]
+    InvalidSignature,
+    #[error("could not unpack fromprior")]
+    InvalidFromPrior,
+    #[error("Could not end relationship")]
+    TargetNotFound,
+    #[error("Could not update connecton")]
+    RepositoryError,
+}
+
+impl RotationError {
```

Comment:
You might consider adding an error code field in the JSON response for each error type to make it easier for clients to handle specific errors programmatically.
e.g
```rust
 impl RotationError {
    pub fn json(&self) -> Json<Value> {
        let (code, message) = match self {
            RotationError::DeserializationError => (1001, "Could not deserialize from prior"),
            RotationError::UnknownIssuer => (1002, "Could not rotate did unknown issuer"),
            RotationError::InvalidSignature => (1003, "Invalid jwt signature on FromPrior value"),
            RotationError::InvalidFromPrior => (1004, "Could not unpack fromprior"),
            RotationError::TargetNotFound => (1005, "Could not end relationship"),
            RotationError::RepositoryError => (1006, "Could not update connection"),
        };

        Json(json!({
            "error": message,
            "code": code
        }))
    }
}
 ```

Reply:
it's a been funny and somehow interesting but the main use of codes is that they are universally known to represent something but in this case we have our custom errors we created giving it codes makes no sense

Reply:
Okay I see